### PR TITLE
DROOLS-1502 Fix incorrect "local" namespace resolution

### PR DESCRIPTION
**Pull request by tarilabs**
Fix `DMNCompilerImpl.resolveTypeRef()` which was not considering local namespace for the QName as resolved by the unmarshalling

Also fix `DMNEvaluatorCompiler.compileDecisionTable` was using wrong "baseline" node for local namespace resolution when calling `resolveType` for the InputClause

### CODE REVIEW

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNCompilerImpl.java [hunk 1]
```diff
@@ -395,14 +397,21 @@ public DMNType resolveTypeRef(DMNModelImpl dmnModel, DMNNode node, NamedElement
                                        null,
                                        Msg.UNKNOWN_TYPE_REF_ON_NODE,
                                        typeRef.toString(),
-                                       localElement.getParentDRGElement().getIdentifierString() );
+                                       Optional.ofNullable(localElement.getParentDRGElement()).map(x->x.getIdentifierString()).orElse(null) );
```
Comment (etirelli) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNCompilerImpl.java]: @tarilabs I am not understanding this? getParentDRGElement should never return null, unless it is used on "<definitions>" or "<itemDefinition>" elements... what is the case here?

File: kie-dmn/kie-dmn-core/src/test/resources/org/kie/dmn/core/actualInputMatchInputValues-forTypeRef-localxmlns.dmn [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    >
+    <!--TEST: do not define feel namespace globally, e.g.: NO root element xmlns:feel="http://www.omg.org/spec/FEEL/20140401" -->
+  <itemDefinition label="MyType" name="MyType">
+    <typeRef xmlns:feel="http://www.omg.org/spec/FEEL/20140401">feel:string</typeRef>
+    <allowedValues constraintsType="enumeration">
+      <text>"a","b","c"</text>
+    </allowedValues>
+  </itemDefinition>
+  <inputData id="_3d560678-a126-4654-a686-bc6d941fe40b" name="MyInput">
+    <variable id="_053333df-1777-45f1-a6c7-56562fbdfdae" name="MyInput" typeRef="MyType"/>
```
Comment (etirelli) [inline kie-dmn/kie-dmn-core/src/test/resources/org/kie/dmn/core/actualInputMatchInputValues-forTypeRef-localxmlns.dmn]: I don't think this typeRef is correct.

File: kie-dmn/kie-dmn-core/src/test/resources/org/kie/dmn/core/actualInputMatchInputValues-forTypeRef-localxmlns.dmn [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  </inputData>
+  <decision id="_497a5306-b2e8-4945-b8b5-82af2e2b99b5" name="MyDecision">
+    <variable id="_514d6d8d-5329-44fa-af91-a7e7addbadd8" name="MyDecision" typeRef="MyType"/>
```
Comment (etirelli) [inline kie-dmn/kie-dmn-core/src/test/resources/org/kie/dmn/core/actualInputMatchInputValues-forTypeRef-localxmlns.dmn]: Same problem here, the typeRef does not look correct.

**Review by etirelli — CHANGES_REQUESTED** [review]
Please see my e-mail @tarilabs.

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNCompilerImpl.java (hunk 1 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNCompilerImpl.java]: This was my mistake.

**Comment by tarilabs** [comment]
@etirelli noted your email explaination. I agree the DMN file is incorrect. 

However, besides of the suprious message line code mistake, I'm quite sure the other corrections I applied to the code are needed.
I will try to make them exhibit with another DMN model file.

**Comment by tarilabs** [comment]
To recap, as per jira DROOLS-1502 I was originally suspecting two distinct issues.

The first about `DMNCompilerImpl.resolveTypeRef()` is actually not an issue, because when the QName will come with the `XMLConstants.NULL_NS_URI` it will actually be resolved as usual via `DMNCompilerImpl.resolveTypeRef()` and recursing up the DMN model tree it will be resolved correctly as per usual `DMNModelInstrumentedBase,getNamespaceURI()` mechanism.
I demonstrate this with file `drools1502-noprefix.dmn` in commits 86841a2 and 86841a2 .

The second issue about `DMNEvaluatorCompiler.compileDecisionTable` is confirmed, I demonstrate it with file `drools1502-InputExpression.dmn` and the stacktrace error with commit f4b1cb6 .
Then I fix it accordingly with commit 0046d1f .

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNEvaluatorCompiler.java [hunk 4]
```diff
@@ -250,7 +250,7 @@ private DMNExpressionEvaluator compileDecisionTable(DMNCompilerContext ctx, DMNM
                                                    ++index );
             } else if ( ic.getInputExpression().getTypeRef() != null ) {
                 QName inputExpressionTypeRef = ic.getInputExpression().getTypeRef();
-                BaseDMNTypeImpl typeRef = (BaseDMNTypeImpl) model.getTypeRegistry().resolveType(resolveNamespaceForTypeRef(inputExpressionTypeRef, ic), inputExpressionTypeRef.getLocalPart());
+                BaseDMNTypeImpl typeRef = (BaseDMNTypeImpl) model.getTypeRegistry().resolveType(resolveNamespaceForTypeRef(inputExpressionTypeRef, ic.getInputExpression()), inputExpressionTypeRef.getLocalPart());
```
Comment (etirelli) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNEvaluatorCompiler.java:254]: good catch!

**Review by etirelli — APPROVED** [review]

**Comment by etirelli** [comment]
+1, looks good!

**Comment by tarilabs** [comment]
[ I needed to rebase ]

**Comment by tarilabs** [comment]
```
Test Result (1 failure / +1)
org.drools.compiler.test.PositionalTest.testPositionalWithNull
```

**Pull request merged**